exploration-module/src/main/java/io/edpn/backend/exploration/adapter/config/KafkaConfig.java

```diff
@@ -0,0 +1,125 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration("ExplorationModuleKafkaConfig")
+public interface KafkaConfig {
+
+    @EnableKafka
+    @Configuration("ExplorationModuleEddnJsonKafkaConsumerConfig")
+    @Slf4j
+    class EddnJsonKafkaConsumerConfig {
+
+        @Value(value = "${exploration.spring.kafka.bootstrap-servers}")
+        private String bootstrapServers;
+
+        public ConcurrentKafkaListenerContainerFactory<String, JsonNode> kafkaListenerContainerFactory(String groupId, CommonErrorHandler errorHandler) {
+            ConcurrentKafkaListenerContainerFactory<String, JsonNode> factory = new ConcurrentKafkaListenerContainerFactory<>();
+            factory.setCommonErrorHandler(errorHandler);
+            factory.getContainerProperties().setAckMode(ContainerProperties.AckMode.RECORD);
+            factory.setConsumerFactory(consumerFactory(groupId));
+            return factory;
+        }
+
+        public ConsumerFactory<String, JsonNode> consumerFactory(String groupId) {
+            Map<String, Object> configProps = new HashMap<>();
+            configProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+            configProps.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+            configProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+            configProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+            configProps.put(JsonDeserializer.VALUE_DEFAULT_TYPE, JsonNode.class);
+            configProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+            configProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+            return new DefaultKafkaConsumerFactory<>(configProps);
+        }
+
+        @Bean(name = "explorationModuleKafkaListenerContainerFactory")
+        public ConcurrentKafkaListenerContainerFactory<String, JsonNode> kafkaListenerContainerFactory(EddnJsonKafkaConsumerConfig kafkaConfig, CommonErrorHandler errorHandler) {
+            return kafkaConfig.kafkaListenerContainerFactory("explorationModule", errorHandler);
+        }
+
+        @Bean(name = "explorationKafkaErrorHandler")
+        public DefaultErrorHandler errorHandler(
+                @Value(value = "${exploration.kafka.backoff.interval:1000}") final int interval,
+                @Value(value = "${exploration.kafka.backoff.multiplier:2000}") final int multiplier,
+                @Value(value = "${exploration.kafka.backoff.max_interval:32000}") final int maxInterval
+        ) {
+            ExponentialBackOff backOff = new ExponentialBackOff();
+            backOff.setInitialInterval(interval);
+            backOff.setMultiplier(multiplier);
+            backOff.setMaxInterval(maxInterval);
+            return new DefaultErrorHandler((consumerRecord, exception) -> {
+                // TODO should there be extra logic to execute when all the retry attempts are exhausted, or do we just drop the message?
```

Comment:
My 2c here: If the client cares, they can retry on timeout. Meaning we're safe to just drop the message.

For example, it may have become corrupt during transmission some how, no reason for us to consume resources in a loop.